lib/components/form/settings.vue

```diff
@@ -10,17 +10,79 @@ including this file, may be copied, modified, propagated, or distributed
 except according to the terms contained in the LICENSE file.
 -->
 <template>
-  <p>Not yet implemented.</p>
+  <div id="form-settings">
+    <alert v-bind="alert" @close="alert.state = false"/>
+    <div class="row">
+      <div class="col-xs-8">
+        <form-edit :form="form" @alert="setAlert"/>
+      </div>
+      <div class="col-xs-4">
+        <div class="panel panel-simple-danger">
+          <div class="panel-heading">
+            <h1 class="panel-title">Danger Zone</h1>
+          </div>
+          <div class="panel-body">
+            <p class="text-center">
+              <button :disabled="awaitingResponse" type="button"
+                class="btn btn-danger" @click="deleteForm.state = true">
+                Delete this form
+              </button>
+            </p>
+          </div>
+        </div>
+      </div>
+    </div>
+    <form-delete :state="deleteForm.state" :form="form"
+      @hide="deleteForm.state = false" @success="afterDelete"/>
+  </div>
 </template>
 
 <script>
+import FormDelete from './delete.vue';
+import FormEdit from './edit.vue';
+import alert from '../../mixins/alert';
+import modal from '../../mixins/modal';
+
 export default {
   name: 'FormSettings',
+  components: { FormEdit, FormDelete },
+  mixins: [
+    alert(),
+    modal('deleteForm')
+  ],
   props: {
     form: {
       type: Object,
       required: true
     }
+  },
+  data() {
+    return {
+      alert: alert.blank(),
+      deleteForm: {
+        state: false
+      }
+    };
+  },
+  methods: {
+    setAlert(alertObj) {
+      this.$emit('alert');
+      this.alert = alertObj;
+    },
+    afterDelete() {
+      const name = this.form.name != null
```

Comment:
i'm guessing this is a common pattern? is it worth centralizing?

Reply:
It's definitely common, but I haven't quite figured out how to centralize it yet. In templates, where there are more space constraints, I've used `form.name || form.xmlFormId`, but while `||` seems safe in this case, I've tended to avoid it outside templates, since it relies on JavaScript's truthiness conditions. A utility function feels a little heavy-handed, and for better or for worse, I've been reluctant to change the data returned by the server. In the forms table, we render slightly different content based on whether there's a form name, so I'd be reluctant to change the `name` property itself. Maybe I could add a `nameOrId` property? Do any of these solutions seem reasonable to you? Do other ideas come to mind?